connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -119,22 +143,27 @@ public void start() {
         consumer = createConsumer();
 
         List<TopicPartition> partitions = new ArrayList<>();
+        TopicDescription topicDesc = topicSupplier.getOrCreateTopic(topic);
+        if (topicDesc != null) {
+            for (Map.Entry<Integer, TopicPartitionInfo> entry : topicDesc.partitions().entrySet()) {
```

Comment:
Actually, the `TopicAdmin` method will always run, and it logs what it does even when it can't use the `AdminClient`. That means no additional log messages here should be necessary.